Update python-package.yml

andreasprlic opened the pull request:
I think we need this change to fix the deploy action

andreasprlic committed:
Update python-package.yml

I think we need this change to fix the deploy action

jsstevenson commented:
@korikuzma maybe this is a good time to add the trusted publishing component from the example repo?

korikuzma commented:
@jsstevenson I agree. @reece would you be able to add trusted publishing on pypi?

korikuzma commented:
@andreasprlic trusted publishing has been setup on pypi. Would you be able to update the file using the example [here](https://github.com/biocommons/biocommons.example/blob/main/.github/workflows/python-package.yml)? You'll only need to change this line: https://github.com/biocommons/biocommons.example/blob/main/.github/workflows/python-package.yml#L129

bsgiles73 commented:
@reece I was not able to reproduce the local unit test error that you ran into in today's collective coding meeting.
```
(3.11) sgiles-MD6M:bioutils shane.giles$ make test
Using Python 3.11
pytest
============================================================================================= test session starts =============================================================================================
platform darwin -- Python 3.11.4, pytest-7.4.4, pluggy-1.5.0 -- /Users/shane.giles/locus/opt/repos/biocommons/bioutils/venv/3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/shane.giles/locus/opt/repos/biocommons/bioutils
configfile: pyproject.toml
testpaths: tests
plugins: optional-tests-0.1.1, cov-4.1.0
collected 22 items

tests/test_normalize.py::test_trim[normalize_fn0] PASSED
tests/test_normalize.py::test_trim[normalize_fn1] PASSED
tests/test_normalize.py::test_anchor[normalize_fn0] PASSED
tests/test_normalize.py::test_anchor[normalize_fn1] PASSED
tests/test_normalize.py::test_trinuc PASSED
tests/test_normalize.py::test_bounds PASSED
tests/test_normalize.py::test_no_trim_no_shuffle PASSED
tests/test_normalize.py::test_shuffle_no_trim PASSED
tests/test_normalize.py::test_mode_string PASSED
tests/test_normalize.py::test_input_alleles_not_modified PASSED
tests/test_normalize.py::test_error_distinct[normalize_fn0] PASSED
tests/test_normalize.py::test_error_distinct[normalize_fn1] PASSED
tests/test_normalize.py::test_error_ref_allele PASSED
tests/test_normalize.py::test_error_vcf_mode_no_trim PASSED
tests/test_seqfetcher.py::test_fetch_seq PASSED
tests/test_seqfetcher.py::test_fetch_seq_ncbi_invalid_positions PASSED
tests/test_seqfetcher.py::test_add_eutils_api_key PASSED
tests/test_seqfetcher.py::test_fetch_seq_errors PASSED
tests/test_seqfetcher.py::test_rate_limit PASSED
tests/test_sequences.py::test_translate_examples PASSED
tests/test_sequences.py::test_translate_selenoproteins PASSED
tests/test_sequences.py::test_translate_vertebrate_mitochondrial PASSED/Users/shane.giles/locus/opt/repos/biocommons/bioutils/venv/3.11/lib/python3.11/site-packages/coverage/control.py:894: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")


---------- coverage: platform darwin, python 3.11.4-final-0 ----------
Name                              Stmts   Miss  Cover
-----------------------------------------------------
src/bioutils/__init__.py              3      3     0%
src/bioutils/_versionwarning.py       7      7     0%
src/bioutils/accessions.py           36     36     0%
src/bioutils/assemblies.py           21     21     0%
src/bioutils/coordinates.py           7      7     0%
src/bioutils/cytobands.py            14     14     0%
src/bioutils/digest.py               25     25     0%
src/bioutils/digests.py              28     28     0%
src/bioutils/exceptions.py            2      2     0%
src/bioutils/normalize.py           131    131     0%
src/bioutils/seqfetcher.py           71     71     0%
src/bioutils/sequences.py           101    101     0%
src/bioutils/vmc_digest.py           11     11     0%
-----------------------------------------------------
TOTAL                               457    457     0%


============================================================================================= 22 passed in 1.94s ==============================================================================================
```

korikuzma commented:
@bsgiles73 same

andreasprlic committed:
feat(CI/CD): updating release workflow based on new template

andreasprlic commented:
@korikuzma I replaced the python-package.yml with the one from the template and updated the pypi bioutils URL accordingly. It looks like it needs a few more tweaks though.

andreasprlic committed:
fix(ruff): fix things ruff is complaining about

andreasprlic committed:
fix(ruff): fix things ruff is complaining about

andreasprlic committed:
fix(CICD): trying to get unit tests to run again

korikuzma reviewed (changes requested):
Oh my bad, didn't see tests are failing now

andreasprlic committed:
fix(CICD): trying to get unit tests to run again

andreasprlic committed:
fix(CICD): trying to get unit tests to run again

andreasprlic committed:
fix(CICD): trying to get unit tests to run again, making Makefile also more in sync with example repo

andreasprlic committed:
fix(CICD): trying to get unit tests to run again

andreasprlic commented:
@korikuzma  after a few more tweaks (in Makefile), tests are passing.

korikuzma commented on .github/workflows/python-package.yml:
```diff
@@ -1,47 +1,48 @@
       run: |
         python -m pip install --upgrade pip
-        pip install --use-deprecated=legacy-resolver -e .[dev]
+        pip install --use-deprecated=legacy-resolver -e .[dev,test]
```
It might be good to consider breaking out test vs dev deps in the example repo

jsstevenson replied:
draft PR: https://github.com/biocommons/biocommons.example/pull/53
